src/ldp/nn/agent/simple_local_agent.py

```diff
@@ -42,6 +46,10 @@ class AgentLMConfig(_LMConfig):
         ),
         validate_default=True,
     )
+    max_traj_token_count: int | None = Field(
+        default=None,
+        description="If set, raise an error if the total tokens in the trajectory exceed this value.",
+    )
```

Comment:
Can we call this `max_messages_token_count` or something? The messages passed in are not necessarily the full trajectory (i.e. if we are collapsing previous observations)